beesdoo_shift/models/cooperative_status.py

```diff
@@ -410,6 +429,52 @@ def _compute_next_countdown_date(self):
         for rec in self:
             rec.next_countdown_date = False
 
+    @api.multi
+    @api.depends(
+        "cooperator_id.shift_shift_ids",
+        "cooperator_id.shift_shift_ids.start_time",
+        "future_alert_date",
+        "today",
+    )
+    def _compute_next_shift(self):
+        # try to use "today" field from status if possible
+        todays = self.mapped("today")
+        if len(todays) == 1:
+            today = todays.pop()
+            now = datetime(today.year, today.month, today.day, 0, 0, 0)
+        else:
+            now = datetime.now()
+
+        # avoid searching for shift in loop
+        cooperator_ids = self.mapped("cooperator_id").ids
+        next_shifts = self.env["beesdoo.shift.shift"].search(
+            [("start_time", ">=", now), ("worker_id", "in", cooperator_ids)],
+            order="worker_id, start_time",
+        )
+
+        next_shift_by_worker = {}
+        for shift in next_shifts:
+            #  take fist shift for each worker
+            if shift.worker_id.id not in next_shift_by_worker:
+                next_shift_by_worker[shift.worker_id.id] = shift
```

Comment:
```suggestion
            next_shift_by_worker.setdefault(shift.worker_id.id, shift)
```

Reply:
Although one liners are nice, I think we lose in readability with setdefault (I had to lookup the function doc).

Reply:
Ok let's see if my brain gets used to it.